client/state/audio/middleware.js

```diff
@@ -0,0 +1,64 @@
+
+/**
+ * Internal dependencies
+ */
+import {
+	HAPPYCHAT_RECEIVE_EVENT,
+	AUDIO_PLAY,
+} from 'state/action-types';
+import { audioPlay } from './actions';
+
+// Maps the audio reference to the audio file source.
+// In future we might want to store this in Redux if
+// we want the sounds to be configurable.
+const REFERENCE_MAP = {
+	'happychat-message-received': '/calypso/audio/chat-pling.wav',
+};
+
+/**
+ * Plays a sound when a message is received on Happychat
+ * @param {function} dispatch Redux dispatcher
+ * @param {Object} action Redux action
+ */
+export const dispatchHappychatSound = ( dispatch, { event } ) => {
+	// Don't play a sound when the customer sends a message
+	if ( event.source === 'customer' ) {
+		return;
+	}
+
+	dispatch( audioPlay( 'happychat-message-received' ) );
+};
+
+export const playSound = ( dispatch, { reference } ) => {
+	if ( typeof Audio !== 'function' ) {
+		// No Audio support in this browser
+		return;
+	}
+
+	const audio = new Audio( REFERENCE_MAP[ reference ] );
+	audio.play();
+};
+
+/**
+ * Action Handlers
+ */
+
+export const handlers = {
+	// Actions which trigger a sound
+	[ HAPPYCHAT_RECEIVE_EVENT ]: dispatchHappychatSound,
```

Comment:
Good question @omarjackman! I could see it either way depending on how much logic is involved. In this case I'd probably lean on putting all of the audio-related actions into this middleware file since it will probably remain small and to-the-point.

@jordwest I may however recommend eliminating the separate function call and dispatch for HappyChat here. is there a need to do that? since the bell itself is a side-effecting response to another action I don't think we need to create yet another action.

```js
const playSound = soundName => () => {
	const url = get( soundNames, soundName );
	if ( ! url ) {
		return;
	}
	const soundClip = new Audio( url );
	soundClip.play();
}

export const handlers = {
	[ HAPPYCHAT_RECEIVE_EVENT ]: playSound( 'happychat-message-received' ),
}
```

I see now that I forgot a logic check…

```js
const playSound = soundName => …

const playHappyChatDing = ( { event } ) =>
	event.source !== 'customer' && playSound( 'happychat-message-received' );

export const handlers = {
	[ HAPPYCHAT ]: playHappyChatDing,
}
```